producers/mysql-producer/src/main/java/com/flipkart/aesop/runtime/producer/avro/MysqlAvroEventManager.java

```diff
@@ -137,19 +141,41 @@ public int createAndAppendEvent(DbChangeEntry changeEntry, DbusEventBufferAppend
 		{
 			final long timestampInNanos = eventHeader.getTimestamp() * 1000000L;
 			final boolean isReplicated = false;
-			for (Row row : rowList)
+			for (Pair pair : listOfPairs)
 			{
-				List<Column> columns = row.getColumns();
+				Row oldRow = (Row) pair.getBefore();
+				Row newRow = (Row) pair.getAfter();
 				// getting the appropriate bin log mapper for the logicalSource
 				BinLogEventMapper<T> binLogEventMapper =
-				        binLogEventMappers.get(lSourceId) == null ? new DefaultBinLogEventMapper<T>(new ORToAvroMapper())
-				                : binLogEventMappers.get(lSourceId);
+						binLogEventMappers.get(lSourceId) == null ? new DefaultBinLogEventMapper<T>(new ORToAvroMapper())
+								: binLogEventMappers.get(lSourceId);
+				String oldValueMapField = AvroSchemaHelper.getRowChangeField(schema);
```

Comment:
I think that schema should be your single source of truth to define whether old value is necessary or not. If we have a flag, then that becomes another source of truth. The implication of this is that when a client receives an event with _oldValue=null, he does not know whether it came as null because of the flag getting turned off, or there was no change at all. This inference can be done by looking at the event type, but it is another thing that the consumer has to be explicitly aware of and handle. Now this inference might be wrong as well because the old value stopped coming because of a bug. I would rather prefer that the schema is explicitly used for communicating this intent.

For schema generation, pass the old value field name to the tool as well. If the field name is null, don't generate the necessary field. If it is not null, then generate the field. Also, add validations to ensure that there are no clashes with existing columns.